Makefile

```diff
@@ -300,13 +300,18 @@ quickstart: $(KUSTOMIZE) manifests #EXHELP Generate the installation release man
 ##@ Docs
 
 .PHONY: crd-ref-docs
-API_REFERENCE_FILENAME := operator-controller-api-reference.md
-crd-ref-docs: $(CRD_REF_DOCS)
-	rm -f $(ROOT_DIR)/docs/refs/api/$(API_REFERENCE_FILENAME)
+OPERATOR_CONTROLLER_API_REFERENCE_FILENAME := operator-controller-api-reference.md
+CATALOGD_API_REFERENCE_FILENAME := catalogd-api-reference.md
+crd-ref-docs: tidy $(CRD_REF_DOCS)
+	$(Q)go mod vendor
+	rm -f $(ROOT_DIR)/docs/refs/api/$(OPERATOR_CONTROLLER_API_REFERENCE_FILENAME)
 	$(CRD_REF_DOCS) --source-path=$(ROOT_DIR)/api \
-	--config=$(ROOT_DIR)/config/base/crd/bases/olm.operatorframework.io_clusterextensions.yaml \
-	--renderer=markdown \
-	--output-path=$(ROOT_DIR)/docs/refs/api/$(API_REFERENCE_FILENAME)
+	--config=$(ROOT_DIR)/docs/refs/api/crd-ref-docs-gen-config.yaml \
```

Comment:
I'd misunderstood the usage of this flag when I first implemented this, but turns out if you don't specify `--source-path` and _do_ specify `--config` as your CRD file, it generates the same doc. Confusing? Or maybe it finds the `api` folder automatically if not specified?

Regardless, this is the actual intended usage. `--config` specifies a configuration file for the generator. At the moment we just specify the reference k8s version, which also consequently fixes the broken k8s API reference links in our doc.

Reply:
From what I can tell, no, the `api/` folder is the only available source of truth for this tool. If the `--source-path` isn't specified, it'll crawl through your folder structure and find the `api/` folder itself.

Do you have an example of what kind of change we would make from a patch like that? Something like field descriptions, validations, or other things not required to be present in the `_types.go` file is what I was thinking of. We may be able to cover this kind of thing using their custom markers and templates ([reference](https://github.com/elastic/crd-ref-docs?tab=readme-ov-file#advanced-features)).

Reply:
Here's an example from catalogd: https://github.com/operator-framework/catalogd/blob/main/config/base/crd/patches/catalog_validation.yaml